crates/contract/src/tee/tee_state.rs

```diff
@@ -338,30 +338,51 @@ impl TeeState {
             .get(tls_public_key)
             .map(|(node_id, _)| node_id.clone())
     }
-    pub fn is_caller_an_attested_node(&self) -> bool {
+    /// Returns true if the caller has at least one participant entry
+    /// whose TLS key matches an attested node belonging to the caller account.
+    ///
+    /// Handles multiple participants per account and supports legacy mock nodes.
+    pub fn is_caller_an_attested_participant(&self, participants: &Participants) -> bool {
         let signer_pk = env::signer_account_pk();
         let signer_id = env::signer_account_id();
-        //let predecessor = env::predecessor_account_id(); //TODO (#1288)  check what is better predecessor or signer
```

Comment:
_note to any reviewers: will be handled in https://github.com/near/mpc/pull/1314_